### PR TITLE
+htc #18925 Add parallelism param to IncConn.handleWithAsyncHandler

### PR DESCRIPTION
**Pull request by hseeberger**
Also fix some doco.

### CODE REVIEW

**Comment by hseeberger** [comment]
See 18925.

**Comment by drewhk** [comment]
OK TO TEST

**Comment by drewhk** [comment]
I am a bit worried about this. The parallelism here does not mean what a user might think it means. Maybe we should find a different name.

**Comment by drewhk** [comment]
LGTM

**Comment by 2beaucoup** [comment]
prefix should be `+htp #18925` (or maybe `!`)

**Comment by hseeberger** [comment]
@2beaucoup, why "htp"? And why should this be a breaking change?

**Comment by drewhk** [comment]
Btw, shouldn't the Java API changed as well?

**Comment by drewhk** [comment]
OK TO TEST

**Comment by 2beaucoup** [comment]
> @2beaucoup, why "htp"?

It's "htc" actually. (htp -> akka-http, htc -> akka-http-core)

> And why should this be a breaking change?

You are right. Should be safe.

**Comment by hseeberger** [comment]
@2beaucoup, I followed the instructions under https://github.com/akka/akka/blob/master/CONTRIBUTING.md. Where are your prefixing rules documented?

**Comment by hseeberger** [comment]
@drewhk, I knew you would ask for that ;-)
Will add the Java stuff. Sigh!

**Comment by hseeberger** [comment]
OK, added Java stuff and changed the prefix to htc.

**Comment by 2beaucoup** [comment]
> Where are your prefixing rules documented?

Currently only in the commit history. ;)
Now we just have to get rid of that nasty double '#'...

File: akka-http-core/src/main/scala/akka/http/javadsl/IncomingConnection.scala [hunk 1]
```diff
@@ -42,15 +42,19 @@ class IncomingConnection private[http] (delegate: akka.http.scaladsl.Http.Incomi
...
 
   /**
    * Handles the connection with the given handler function.
-   * Returns the materialization result of the underlying flow materialization.
    */
   def handleWithAsyncHandler(handler: Function[HttpRequest, Future[HttpResponse]], materializer: Materializer): Unit =
     delegate.handleWithAsyncHandler(handler.apply(_).asInstanceOf[Future[sm.HttpResponse]])(materializer)
+
+  /**
+   * Handles the connection with the given handler function.
```
Comment (2beaucoup) [inline akka-http-core/src/main/scala/akka/http/javadsl/IncomingConnection.scala]: This should mention the new param.

Comment (hseeberger) [inline akka-http-core/src/main/scala/akka/http/javadsl/IncomingConnection.scala]: The other params aren't documented neither.

Comment (drewhk) [inline akka-http-core/src/main/scala/akka/http/javadsl/IncomingConnection.scala]: boy scout ;)

**Comment by hseeberger** [comment]
OK, all unnecessary #s deleted.

**Comment by ktoso** [comment]
LGTM, thanks Heiko

**Comment by ktoso** [comment]
AFAR this is binary incompatible, but we don't mind during Milestones and there's other changing stuff anyway during the rush towards 2.0 ;-)

**Pull request merged**